oak_functions/loader/src/metrics.rs

```diff
@@ -120,19 +118,16 @@ impl OakApiNativeExtension for PrivateMetricsExtension<Logger> {
 /// Provides logic for the host ABI function [`report_metric`](https://github.com/project-oak/oak/blob/main/docs/oak_functions_abi.md#report_metric).
 fn report_metric(
     extension: &mut PrivateMetricsExtension<Logger>,
-    raw_label: Vec<u8>,
-    value: i64,
+    request: Vec<u8>,
 ) -> Result<(), OakStatus> {
-    let label = std::str::from_utf8(raw_label.as_slice()).map_err(|err| {
-        extension.log_warning(&format!(
-            "report_metric(): Not a valid UTF-8 encoded string: {:?}\nContent: {:?}",
-            err, raw_label
-        ));
-        OakStatus::ErrInvalidArgs
-    })?;
-    extension.log_debug(&format!("report_metric(): {}", label));
-    extension.report_metric(label, value).map_err(|err| {
-        extension.log_error(&format!("report_metric(): {:?}", err));
-        OakStatus::ErrInternal
-    })
+    let request: ReportMetricRequest =
+        bincode::deserialize(&request).expect("Fail to deserialize report metric request.");
```

Comment:
Why are we using bincode instead of proto serialization?
Or it's just to encode and decode Rust structures?

Reply:
Yes, we just use it to decode Rust. I think we'll revisit this when we integrate with #2500